app/jobs/digital_object_import_processing_job.rb

```diff
@@ -44,13 +44,23 @@ def self.perform(digital_object_import_id)
   end
 
   def self.digital_object_for_digital_object_data(digital_object_data)
-    # If a uid is present in the digital_object_data, then this is an update operation
-    return DigitalObject.find_by_uid!(digital_object_data['uid']) if digital_object_data['uid'].present?
-
-    # No uid present, so we'll create a new object with type based on the digital_object_type value
-    return Hyacinth::Config.digital_object_types.key_to_class(digital_object_data['digital_object_type']).new if digital_object_data['digital_object_type'].present?
+    if digital_object_data['uid'].present? && digital_object_data['assign_uid'].present?
+      raise ArgumentError, 'Either "uid" or "assign_uid" attribute may be present in digital_object_data, but not both.'
+    end
+    if digital_object_data['uid'].present?
+      # If a uid is present in the digital_object_data, then this is an update operation
+      DigitalObject.find_by_uid!(digital_object_data['uid'])
+    elsif appropriate_create_data_present(digital_object_data)
+      Hyacinth::Config.digital_object_types.key_to_class(digital_object_data['digital_object_type']).new
+    end
```

Comment:
```suggestion
if digital_object_data['uid'].present? && digital_object_data['assign_uid'].present?
      raise ArgumentError, 'Either "uid" or "assign_uid" attribute may be present in digital_object_data, but not both.'
    end
    return DigitalObject.find_by_uid!(digital_object_data['uid']) if digital_object_data['uid'].present?

    ensure_appropriate_create_data_present!(digital_object_data)
    Hyacinth::Config.digital_object_types.key_to_class(digital_object_data['digital_object_type']).new
```

Reply:
Proposing a minor refactor (with another related comment below) because I thought it wasn't immediately obvious that the elsif condition check might raise an error.  The `ensure_appropriate_create_data_present!` method, ending with a `!` could be a good way to indicate this at a glance.